File: SpringBootApp/src/main/java/com/example/springbootapp/data/dto/AuthResponseDto.java
package com.example.springbootapp.data.dto;

import com.example.springbootapp.data.entities.Enums.Gender;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@NoArgsConstructor
public abstract class AuthResponseDto {
    protected String jwt;
    protected String id;
    protected String username;
    protected String firstName;
    protected String lastName;
    protected String profilePicUrl;
    protected String email;
    protected LocalDate birthDate;
    protected Gender gender;
}

File: ClientAdmin/app/src/main/java/com/example/clientadmin/model/dto/AuthResponseDto.kt
package com.example.clientadmin.model.dto

import com.example.clientadmin.model.enumerator.Gender
import com.squareup.moshi.JsonClass
import java.io.Serializable
import java.time.LocalDate

@JsonClass(generateAdapter = true)
data class AuthResponseDto(
    val jwt: String,
    val username: String,
    val firstName: String,
    val lastName: String,
    val birthDate: LocalDate,
    val email: String,
    val gender: Gender,
    val profilePicUrl: String
): Serializable
